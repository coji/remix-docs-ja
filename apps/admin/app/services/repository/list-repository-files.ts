import fg from 'fast-glob'
import { okAsync } from 'neverthrow'
import { getRepositoryFileContent } from './get-repository-file-content'
import type { RepositoryFile } from './types'

export const listRepositoryFiles = async (
  directory: string,
  option: { pattern: string; excludes: string[] } = {
    pattern: '**/*',
    excludes: [],
  },
) => {
  const filenames = await fg(
    [option.pattern, ...option.excludes.map((e) => `!${e}`)],
    {
      cwd: directory,
      onlyFiles: true,
    },
  )

  const files: RepositoryFile[] = []
  for (const filename of filenames) {
    const result = await getRepositoryFileContent(directory, filename)
    if (result.isOk()) {
      files.push(result.value)
    }
  }

  return okAsync(files)
}
