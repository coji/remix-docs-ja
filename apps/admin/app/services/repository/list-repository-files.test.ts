import path from 'node:path'
import { describe, expect, test } from 'vitest'
import { listRepositoryFiles } from './list-repository-files'

describe('listRepositoryFiles', () => {
  test('should return the list of files in the directory', async () => {
    const directory = path.join('test/fixtures/projects/projectA/docs')
    const files = await listRepositoryFiles(directory)
    expect(files.isOk()).toBe(true)
    expect(files._unsafeUnwrap()).toMatchSnapshot()
  })

  // excludes
  test('should exclude files that match the exclude pattern', async () => {
    const directory = path.join('test/fixtures/projects/projectA/docs')
    const files = await listRepositoryFiles(directory, {
      pattern: '**/*',
      excludes: ['index.*'],
    })
    expect(files.isOk()).toBe(true)
    expect(files._unsafeUnwrap()).toMatchSnapshot()
  })

  test('should return empty array if the directory does not exist', async () => {
    const directory = path.join('test/fixtures/projects/non-existent')
    const result = await listRepositoryFiles(directory)
    expect(result.isOk()).toBe(true)
    expect(result._unsafeUnwrap()).toEqual([])
  })
})
