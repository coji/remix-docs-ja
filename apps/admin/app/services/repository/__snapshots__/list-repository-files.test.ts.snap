// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`listRepositoryFiles > should exclude files that match the exclude pattern 1`] = `
[
  {
    "content": "---
title: About
---

# About this project

This project is a　web-based application designed to simplify the process of managing tasks and projects. It provides a user-friendly interface that allows users to create, update, and track the progress of their tasks. The application also includes features for team collaboration, such as task assignment and comment sections. It is built using modern web technologies, ensuring a responsive and efficient user experience.
",
    "filename": "guide/about.md",
    "md5": "8108d6397ef101da566e35374327cac8",
  },
  {
    "content": "---
title: Introduction to Task Management Application
---

This document provides an overview of our Task Management Application, a web-based tool designed to simplify the process of managing tasks and projects. It offers a user-friendly interface that allows users to create, update, and track the progress of their tasks. The application also includes features for team collaboration, such as task assignment and comment sections. It is built using modern web technologies, ensuring a responsive and efficient user experience.
",
    "filename": "guide/introduction.md",
    "md5": "e94888f5624bcdbe551052c01be8bae2",
  },
]
`;

exports[`listRepositoryFiles > should return the list of files in the directory 1`] = `
[
  {
    "content": "---
title: Getting Started with Task Management Application
---

This document will guide you through the initial setup and basic usage of our Task Management Application. We will cover topics such as creating an account, logging in, creating and managing tasks, and collaborating with your team. By the end of this guide, you should have a good understanding of how to use the application to effectively manage your tasks and projects.
",
    "filename": "index.md",
    "md5": "b6ea662901a38626e3c5f45e532082ad",
  },
  {
    "content": "---
title: About
---

# About this project

This project is a　web-based application designed to simplify the process of managing tasks and projects. It provides a user-friendly interface that allows users to create, update, and track the progress of their tasks. The application also includes features for team collaboration, such as task assignment and comment sections. It is built using modern web technologies, ensuring a responsive and efficient user experience.
",
    "filename": "guide/about.md",
    "md5": "8108d6397ef101da566e35374327cac8",
  },
  {
    "content": "---
title: Introduction to Task Management Application
---

This document provides an overview of our Task Management Application, a web-based tool designed to simplify the process of managing tasks and projects. It offers a user-friendly interface that allows users to create, update, and track the progress of their tasks. The application also includes features for team collaboration, such as task assignment and comment sections. It is built using modern web technologies, ensuring a responsive and efficient user experience.
",
    "filename": "guide/introduction.md",
    "md5": "e94888f5624bcdbe551052c01be8bae2",
  },
]
`;
